api/app/auto_spatial_advisory/classify_hfi.py

```diff
@@ -0,0 +1,48 @@
+""" Proof of concept, run classification on HFI GeoTiff
+"""
+import os
+import numpy as np
+from osgeo import gdal
+
+
+def classify_hfi(source_path, target_path):
+    """
+    Given a source path of some HFI GeoTIFF, classify the GeoTIFF and save it to a new GeoTIFF.
+    The output GeoTIFF will use 8 bit unsigned values.
+
+    NOTE: This could be done in memory!
+    """
+    # Read the source data.
+    source_tiff = gdal.Open(source_path, gdal.GA_ReadOnly)
+    source_band = source_tiff.GetRasterBand(1)
+    source_data = source_band.ReadAsArray()
+    # Classify the data.
+    classified = np.where(source_data < 4000, 0, source_data)
+    classified = np.where((classified >= 4000) & (classified < 10000), 1, classified)
+    classified = np.where(classified >= 10000, 2, classified)
+
+    # Remove any existing target file.
+    if os.path.exists(target_path):
+        os.remove(target_path)
+    output_driver = gdal.GetDriverByName("GTiff")
+    # Create an object with the same dimensions as the input, but with 8 bit unsigned values.
+    target_tiff = output_driver.Create(target_path, xsize=source_band.XSize,
+                                       ysize=source_band.YSize, bands=1, eType=gdal.GDT_Byte)
+    # Set the geotransform and projection to the same as the input.
+    target_tiff.SetGeoTransform(source_tiff.GetGeoTransform())
+    target_tiff.SetProjection(source_tiff.GetProjection())
+
+    # Write the classified data to the band.
+    target_band = target_tiff.GetRasterBand(1)
+    target_band.SetNoDataValue(0)
```

Comment:
Variable-izing this 0 value might be a good idea. In my branch I've been using -10000 as the NoDataValue to remain consistent with the FTL.

Reply:
It's going to have different meanings for different rasters - so I don't know about that. In this case, we're classifying into three sets 0. < 4000 1. >=4000 and <10000, 2. >=10000
With FTL (or other things) 0 may have some different meaning.